File: javascript/ql/src/Security/CWE-1427/examples/prompt-injection.js
const express = require("express");
const OpenAI = require("openai");

const app = express();
const client = new OpenAI();

app.get("/chat", async (req, res) => {
    let persona = req.query.persona;

    // BAD: user input is used directly in a system-level prompt
    const response = await client.chat.completions.create({
        model: "gpt-4.1",
        messages: [
            {
                role: "system",
                content: "You are a helpful assistant. Act as a " + persona,
            },
            {
                role: "user",
                content: req.query.message,
            },
        ],
    });

    res.json(response);
});

File: python/ql/src/Security/CWE-1427/examples/prompt-injection.py
from flask import Flask, request
from openai import OpenAI

app = Flask(__name__)
client = OpenAI()


@app.get("/chat")
def chat():
    persona = request.args.get("persona")

    # BAD: user input is used directly in a system-level prompt
    response = client.chat.completions.create(
        model="gpt-4.1",
        messages=[
            {
                "role": "system",
                "content": "You are a helpful assistant. Act as a " + persona,
            },
            {
                "role": "user",
                "content": request.args.get("message"),
            },
        ],
    )

    return response
